src/main/java/org/kuali/kfs/sys/context/KFSConfigurer.java

```diff
@@ -185,26 +180,47 @@ private void initializeResourceLoaders() throws Exception {
         GlobalResourceLoader.start();
     }
 
-    @Override
-    public List<Lifecycle> loadLifecycles() {
-        final List<Lifecycle> lifecycles = new LinkedList<>();
-
-        // this validation of our service list needs to happen after we've loaded our configs so it's a lifecycle
-        lifecycles.add(new BaseLifecycle() {
-            @Override
-            public void start() throws Exception {
-                super.start();
-            }
-        });
-
-        return lifecycles;
-    }
-
+    // These ApplicationEventS are for the startup ApplicationContext (AC), not the main AC; however, most of/all the
+    // methods herein deal with the main AC.
     @Override
     public void onApplicationEvent(final ApplicationEvent applicationEvent) {
         if (applicationEvent instanceof ContextRefreshedEvent) {
             loadDataDictionary();
             doAdditionalContextStartedLogic();
+
+            final ApplicationContext mainApplicationContext = GlobalResourceLoader.getContext();
+
+            // Publish Step classes as Components. These are not in the
+            // DD so are not published by
+            // DataDictionaryComponentPublisherService.publishAllComponents().
+            publishBatchStepComponents(mainApplicationContext);
+            initDirectories(mainApplicationContext);
+            updateWorkflow(mainApplicationContext);
+            initScheduler(mainApplicationContext);
+            
+            // CU customization
+            if (shouldAllowLocalBatchExecution()) {
```

Comment:
I would suggest moving this custom block into the new `initScheduler()` method.

Reply:
moved